File: 875-koko-eating-bananas/875-koko-eating-bananas.java
class Solution {
    public int minEatingSpeed(int[] p, int h) {
        
        int n=p.length;
        Arrays.sort(p);
        
        int l=1;//r=*max_element(p.begin(),p.end());
        int r=p[n-1];
        while(l<r)
        {
            int m=(l+r)/2;

            int x=0;
            
            for(int i:p)
            {
                x+= (i/m);
                
                if(i%m!=0)
                    x=x+1;
            }
            if(x<=h)
            {
                r=m;
            }
            
            else
            {
                l=m+1;
            }
       
        }
        
        return r;
        
    }
}

File: 875-koko-eating-bananas/875-koko-eating-bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& p, int h) {
        
        int n=p.size();
        
        long long int l=1,r=0;
        
        for(int i=0;i<n;i++)
        {
            r+=p[i];
        }
        
        int ans=0;
        
        while(l<=r)
        {
            long long int m=(l+r)/2;
            
            int c=0;
            
            for(int i=0;i<n;i++)
            {
                c+=p[i]/m;
                
                if(p[i]%m)
                {
                    c++;
                }
            }
            
            if(c<=h)
            {
                ans=m;
                r=m-1;
            }
            else
            {
                l=m+1;
            }
        }
        
        return ans;
    }
};